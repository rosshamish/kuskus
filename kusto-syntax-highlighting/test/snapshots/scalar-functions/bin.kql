// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/binfunction

bin(4.5, 1)

bin(time(16d), 7d)


bin(datetime(1970-05-11 13:45:07), 1d)

T | summarize Hits=count() by bin(Duration, 1s)
