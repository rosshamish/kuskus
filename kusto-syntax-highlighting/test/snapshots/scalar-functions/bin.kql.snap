>// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/binfunction
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>
>bin(4.5, 1)
#^^^ source.kusto meta.scalar.function.bin.kusto support.function.kusto
#   ^ source.kusto meta.scalar.function.bin.kusto
#    ^^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#       ^^ source.kusto meta.scalar.function.bin.kusto
#         ^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#          ^ source.kusto meta.scalar.function.bin.kusto
>
>bin(time(16d), 7d)
#^^^ source.kusto meta.scalar.function.bin.kusto support.function.kusto
#   ^ source.kusto meta.scalar.function.bin.kusto
#    ^^^^ source.kusto meta.scalar.function.bin.kusto storage.type.kusto
#        ^ source.kusto meta.scalar.function.bin.kusto
#         ^^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#            ^ source.kusto meta.scalar.function.bin.kusto
#             ^^ source.kusto meta.scalar.function.bin.kusto
#               ^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#                 ^ source.kusto meta.scalar.function.bin.kusto
>
>
>bin(datetime(1970-05-11 13:45:07), 1d)
#^^^ source.kusto meta.scalar.function.bin.kusto support.function.kusto
#   ^ source.kusto meta.scalar.function.bin.kusto
#    ^^^^^^^^ source.kusto meta.scalar.function.bin.kusto storage.type.kusto
#            ^ source.kusto meta.scalar.function.bin.kusto
#             ^^^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#                 ^ source.kusto meta.scalar.function.bin.kusto
#                  ^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#                    ^ source.kusto meta.scalar.function.bin.kusto
#                     ^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#                       ^ source.kusto meta.scalar.function.bin.kusto
#                        ^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#                          ^ source.kusto meta.scalar.function.bin.kusto
#                           ^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#                             ^ source.kusto meta.scalar.function.bin.kusto
#                              ^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#                                ^ source.kusto meta.scalar.function.bin.kusto
#                                 ^^ source.kusto meta.scalar.function.bin.kusto
#                                   ^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#                                     ^ source.kusto meta.scalar.function.bin.kusto
>
>T | summarize Hits=count() by bin(Duration, 1s)
#^^^^ source.kusto
#    ^^^^^^^^^ source.kusto keyword.other.query.kusto
#             ^^^^^^ source.kusto
#                   ^^^^^^^ source.kusto support.function.kusto
#                          ^ source.kusto
#                           ^^ source.kusto keyword.other.operator.kusto
#                             ^ source.kusto
#                              ^^^ source.kusto meta.scalar.function.bin.kusto support.function.kusto
#                                 ^ source.kusto meta.scalar.function.bin.kusto
#                                  ^^^^^^^^ source.kusto meta.scalar.function.bin.kusto
#                                          ^^ source.kusto meta.scalar.function.bin.kusto
#                                            ^^ source.kusto meta.scalar.function.bin.kusto constant.numeric.kusto
#                                              ^ source.kusto meta.scalar.function.bin.kusto
>